snippets/getColonTime.md

```diff
@@ -0,0 +1,13 @@
+### toColonTime
+
+Returns transformed time integers from provided date object to "colon time" representation. Formats hour to 12 hour and maintains digit length of colon time format (HH:MM:SS).
+
+```js
+let toColonTime = ( date ) => {
+    let times = [ date.getHours(), date.getMinutes(), date.getSeconds() ];
+    times[0] = times[0] === 0 || times[0] === 12 || times[0] === 24 ? 12 : times[0] % 12;
+    
+    return times.map( time => time.toString().padStart(2, "0") ).join(":");
+}
+// toColonTime12(new Date()) -> "08:38:00"
```

Comment:
The examples should be in a different js block. Please update your snippets to match

# Example

> ### toColonTime
> 
> Returns transformed time integers from provided date object to "colon time" representation. Formats hour to 12 hour and maintains digit length of colon time format (HH:MM:SS).
> ```js
> const toColonTime = ( date ) => {
>     let times = [ date.getHours(), date.getMinutes(), date.getSeconds() ];
>     times[0] = times[0] === 0 || times[0] === 12 || times[0] === 24 ? 12 : times[0] % 12;
>     
>     return times.map( time => time.toString().padStart(2, "0") ).join(":");
> }
> ```
> ```js
> toColonTime12(new Date()) -> "08:38:00"
> ```

Reply:
I've realized I can reduce this function to
```js
const getColonTimeFromDate = date => date.toTimeString().slice(0, 8)
```

Not sure if it's worth submitting at all.

Reply:
Worth submitting IMHO; some don't realize one can just slice.
Keep in mind that many program JavaScript who are still novices; either new to programming or program other languages.  Having a copy/paste reference is handy!

Reply:
I had originally preferred constructing the colon time string from individual values because it allowed you to format those values beforehand such as formatting the hours to 12 hour format. To format this you'd need to format the result of the function. I felt it was more functional the other way, but this would still work. I imagine this function would be an 'endpoint' though, so rather than have this function encapsulate parameters being formatted, you'd format the result of the function.